src/core/public/http/fetch.ts

```diff
@@ -160,11 +182,31 @@ export class Fetch {
       throw new HttpFetchError(response.statusText, request, response, body);
     }
 
-    return new HttpResponse({ request, response, body });
+    return { fetchOptions, request, response, body };
   }
 
-  private shorthand(method: string) {
-    return (path: string, options: HttpFetchOptions = {}) =>
-      this.fetch(path, { ...options, method });
+  private shorthand(method: string): HttpHandler {
+    return (pathOrOptions: string | HttpFetchOptionsWithPath, options?: HttpFetchOptions) => {
+      const optionsWithPath = validateFetchArguments(pathOrOptions, options);
+      return this.fetch({ ...optionsWithPath, method });
+    };
```

Comment:
The validation address the issue.

Still wonder if we could add proper typing to the handler ?

Would this work?

```typescript
<T extends string | HttpFetchOptionsWithPath>(pathOrOptions: T, options: T extends string ? HttpFetchOptions : never)
```

Reply:
`HttpHandler` type overloads the signature. https://github.com/elastic/kibana/pull/53734/files#diff-d1fdf4d9c9d69c4fdfc4b6963e9050cfR254

it's not immediately obvious 
- https://github.com/elastic/kibana/pull/53734/files#diff-fc8d4853dbfb5a62d938e7155ee7d372R188
- https://github.com/elastic/kibana/pull/53734/files#diff-fc8d4853dbfb5a62d938e7155ee7d372R68